source/agora/consensus/EnrollmentManager.d

```diff
@@ -936,17 +923,85 @@ unittest
 
     auto utxo_hash = utxo_hashes[0];
     Enrollment enroll;
-    assert(man.createEnrollment(utxo_hash, enroll));
+    assert(man.createEnrollment(utxo_hash, 1, enroll));
     assert(man.add(0, &storage.findUTXO, enroll));
     assert(man.hasEnrollment(utxo_hash));
 
     PreimageInfo result_image;
     assert(man.getValidatorPreimage(utxo_hash, result_image));
     assert(result_image == PreimageInfo.init);
-    auto preimage = PreimageInfo(utxo_hash, man.preimages[100], 1100);
+    auto preimage = PreimageInfo(utxo_hash, man.cycle_preimages[100], 1100);
     assert(man.addPreimage(preimage));
     assert(man.getValidatorPreimage(utxo_hash, result_image));
     assert(result_image.enroll_key == utxo_hash);
-    assert(result_image.hash == man.preimages[100]);
+    assert(result_image.hash == man.cycle_preimages[100]);
     assert(result_image.height == 1100);
 }
+
+// test for generating pre-images
+unittest
+{
+    KeyPair key_pair = KeyPair.random();
```

Comment:
Using random value can be deceiving. Because when using random values, you have to program your tests, which can themselves have bugs. To test the most basic properties of a function, we favor fixed value.
In this case you could start with a fixed value for the key and the salt and just compute that the hashes are what you expect.